### PR TITLE
fix(STONEINTG-737): cleanup related resources after ITS is deleted

### PR DESCRIPTION
**Pull request by dirgim**
* If the integrationTestScenario has related ephemeral environments, ensure that those are cleaned up when it is deleted
* Add a finalizer to IntegrationTestScenarios in order to be able to correctly handle the resource cleanup
* Add utility functions to handle the above functionalities
* Update diagrams with the new behavior

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: controllers/scenario/predicates.go [hunk 1]
```diff
@@ -21,15 +21,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-// IntegrationPipelineRunStartedPredicate returns a predicate which filters out
-// only created integration scenarios
+// IntegrationScenarioCreatedPredicate returns a predicate which filters out
```
Comment (jsztuka) [inline controllers/scenario/predicates.go:24]: I feel that the functionname is little bit misleading, can we consider something like: 
_IntegrationScenarioCreatedAndDeletedPredicate_ ?

Comment (dirgim) [reply controllers/scenario/predicates.go:24]: Tbh I was thinking of removing the predicate altogether as we're responding to basically all types of events now.

Comment (dirgim) [reply controllers/scenario/predicates.go]: Ack, updated

Comment (MartinBasti) [reply controllers/scenario/predicates.go]: Do we really want to react to generic events?

File: helpers/integration.go [hunk 2]
```diff
@@ -478,3 +479,35 @@ func AddFinalizerToPipelineRun(adapterClient client.Client, logger IntegrationLo
...
 }
+
+// RemoveFinalizerFromScenario removes the finalizer from the IntegrationTestScenario.
+// If finalizer was not removed successfully, a non-nil error is returned.
+func RemoveFinalizerFromScenario(adapterClient client.Client, logger IntegrationLogger, ctx context.Context, scenario *v1beta1.IntegrationTestScenario, finalizer string) error {
+	patch := client.MergeFrom(scenario.DeepCopy())
+	if ok := controllerutil.RemoveFinalizer(scenario, finalizer); ok {
+		err := adapterClient.Patch(ctx, scenario, patch)
+		if err != nil {
+			return fmt.Errorf("error occurred while patching the updated inte after finalizer removal: %w", err)
```
Comment (hongweiliu17) [inline helpers/integration.go:490]:
```suggestion
			return fmt.Errorf("error occurred while patching the updated IntegrationTestScenario after finalizer removal: %w", err)
```

**Review by hongweiliu17 — APPROVED** [review]
Only one comment for log message. Others looks to me.

File: controllers/scenario/predicates.go (hunk 1 above)
Comment (dirgim) [reply controllers/scenario/predicates.go]: I'm fine with reacting to all events for this owned resource and making sure it's validated tbh

File: helpers/integration.go (hunk 2 above)
Comment (dirgim) [reply helpers/integration.go]: Good catch, thanks. Fixed it now.

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**